clarify deploy-full-rp-service-in-dev.md guide

mbukatov opened the pull request:
### Which issue this PR addresses:

None.

### What this PR does / why we need it:

Improves "Deploy an Entire RP Development Service" guide so that one can avoid problems caused by misunderstanding of the procedure.

### Test plan for issue:

Changes are based on my experience with this procedure. No further testing expected beyond review.

mbukatov commented:
Publishing as a draft for visibility, I still need to polish this a bit. Feedback welcome.

yehornaumenko commented on docs/deploy-full-rp-service-in-dev.md:
```diff
@@ -3,31 +3,56 @@
 
 1. Your development environment is prepared according to the steps outlined in [Prepare Your Dev Environment](./prepare-your-dev-environment.md)
-
+2. During the deployment, don't edit files in your ARO-RP repository so that
+   `git status` reports a clean working tree. Otherwise aro container image
+   will have `-dirty` suffix, which can cause problems esp. when the repository
+   becomes dirty during the procedure.
```
I guess we need to prevent this programmatically in this scenario because it can't be wanted. Unless somebody wanna deploy an RP using some not committed changes.
I would add some checks or warning messages to the scripts we use or _make_'s targets.
But we can do it in another PR. Does it make sense?

jaitaiwan replied:
What sort of problems does the -dirty suffix cause? Is it only an issue when deploying a shared rp or is it an issue for local dev as well?

SrinivasAtmakuri replied:
One solution would be to set COMMIT/TAG = latest if RP_MODE == development, this will make reruns or testing a lot easier.

cadenmarchese replied:
To me it's more of a recommendation. You can definitely deploy with the dirty tag if you want (I do it and sometimes it's an accident but it still works), but you will have the potential issues mentioned above.

Should we just change the wording to "it's recommended to..." and add the content here to the document? https://github.com/Azure/ARO-RP/pull/2964/files#r1238448125

mbukatov replied:
I updated the text as suggested by @cadenmarchese .

mbukatov commented:
The changes are ready for review. I addressed the feedback about the dirty image tag, and tweaked the text about container registry used for `make publish-image*` targets.

tiguelu reviewed (dismissed):
LGTM

UlrichSchlueter commented:
/azp run ci

tiguelu commented:
/azp run ci

mbukatov commented:
Rebased on top of current master (14e3c5a88) to unblock the github checks.

tiguelu commented:
/azp run ci
